src/components/dialog/DialogDismissibleView.tsx

```diff
@@ -54,216 +55,200 @@ interface LocationProps {
 }
 
 const DEFAULT_DIRECTION = PanningProvider.Directions.DOWN;
+const TOP_INSET = Constants.isIphoneX ? Constants.getSafeAreaInsets().top : Constants.isIOS ? 20 : 0;
+const BOTTOM_INSET = Constants.isIphoneX ? Constants.getSafeAreaInsets().bottom : Constants.isIOS ? 20 : 0;
+
+const DialogDismissibleView = (props: Props) => {
+  const {direction = DEFAULT_DIRECTION, visible: propsVisible, containerStyle, style, children, onDismiss} = props;
+  // @ts-expect-error
+  const {isPanning, dragDeltas, swipeDirections} = useContext<PanContextProps>(PanningContext);
+
+  const width = useRef<number>(Constants.screenWidth);
+  const height = useRef<number>(Constants.screenHeight);
+  const thresholdX = useRef<number>(0);
+  const thresholdY = useRef<number>(0);
+  const counter = useRef<number>(0);
+  const containerRef = useRef<typeof View>();
+  const animatedValue = useRef<Animated.AnimatedValue>(new Animated.Value(0));
+  const swipe = useRef<PanDirectionsProps>({});
+  const prevDragDeltas = useRef<PanAmountsProps>();
+  const prevSwipeDirections = useRef<PanDirectionsProps>();
+  const visible = useRef<boolean>(Boolean(propsVisible));
+
+  const getHiddenLocation = useCallback((left: number, top: number): LocationProps => {
+    const result = {left: 0, top: 0};
+    switch (direction) {
+      case PanningProvider.Directions.LEFT:
+        result.left = -left - width.current;
+        break;
+      case PanningProvider.Directions.RIGHT:
+        result.left = Constants.screenWidth - left;
+        break;
+      case PanningProvider.Directions.UP:
+        result.top = -top - height.current - TOP_INSET;
+        break;
+      case PanningProvider.Directions.DOWN:
+      default:
+        result.top = Constants.screenHeight - top + BOTTOM_INSET;
+        break;
+    }
 
-class DialogDismissibleView extends PureComponent<Props, State> {
+    return result;
+  },
+  [direction]);
 
-  public static defaultProps: Partial<Props> = {
-    direction: DEFAULT_DIRECTION,
-    onDismiss: () => {}
-  };
+  const hiddenLocation = useRef<LocationProps>(getHiddenLocation(0, 0));
 
-  private hiddenLocation: LocationProps;
-  private animatedValue = new Animated.Value(0);
-  private width = Constants.screenWidth;
-  private height = Constants.screenHeight;
-  private counter = 0;
-  private swipe: PanDirectionsProps = {};
-  private thresholdX = 0;
-  private thresholdY = 0;
-  private ref = React.createRef<any>();
+  const animateTo = useCallback((toValue: number, animationEndCallback?: Animated.EndCallback) => {
+    Animated.timing(animatedValue.current, {
+      toValue,
+      duration: 300,
+      easing: Easing.bezier(0.2, 0, 0.35, 1),
+      useNativeDriver: true
+    }).start(animationEndCallback);
+  }, []);
 
-  constructor(props: Props) {
-    super(props);
+  const isSwiping = useCallback((): boolean => {
```

Comment:
Also here.. useCallback seem redundant.
Even the function content only use reference values

Reply:
Also, verify you really need to declare a return type. 
Typescript usually handles it automaticaly

Reply:
`isSwiping` is used in `onDrag` which is used in `useEffect`.
Removed a couple of return values.